### PR TITLE
Reopen SMB dialog on failed load list

### LINKED ISSUE #2195: Open retry SMB dialog on list load failure for SMB.
Open retry SMB dialog on list load failure for SMB
Currently SMB will keep loading indefinitely.

### PR DESCRIPTION
**Pull request by VishalNehra**
#### Issue tracker   
Fixes will automatically close the related issue

Fixes #2195 
-or-   
Addresses #

#### Release  
Addresses release/3.5
  
#### Test cases
- [ ] Covered
  
#### Manual testing
- [x] Done  
  
If yes,  
- Device:
- OS:

#### Build tasks success  
Successfully running following tasks on local 
- [x] `./gradlew assembledebug`
- [x] `./gradlew spotlessCheck`

#### Related PR  
Related to PR #

#### Additional Info

### CODE REVIEW

**Review by TranceLove — DISMISSED** [review]
Works fine on LG Nexus 5x running AOSPExtended (9.0). +1.

File: app/src/main/java/com/amaze/filemanager/ui/fragments/MainFragment.java [hunk 1]
```diff
@@ -950,13 +949,9 @@ public void loadlist(final String path, final boolean back, final OpenMode openM
...
-              if (data != null && data.second != null) {
-                boolean isPathLayoutGrid =
-                    dataUtils.getListOrGridForPath(path, DataUtils.LIST) == DataUtils.GRID;
-                setListElements(data.second, back, path, data.first, false, isPathLayoutGrid);
-              } else {
-                AppConfig.toast(requireContext(), getString(R.string.unknown_error));
-              }
+              boolean isPathLayoutGrid =
+                  dataUtils.getListOrGridForPath(path, DataUtils.LIST) == DataUtils.GRID;
+              setListElements(data.second, back, path, data.first, false, isPathLayoutGrid);
```
[the remark this answers is not in the document]
Comment (VishalNehra) [reply app/src/main/java/com/amaze/filemanager/ui/fragments/MainFragment.java:954]: We're handling the null case in setListElements. We'll cancel and navigate to home directory.

[the remark this answers is not in the document]
Comment (TranceLove) [reply app/src/main/java/com/amaze/filemanager/ui/fragments/MainFragment.java:954]: Sorry... looks like not.

Tested using Fairphone 3 running LineageOS 16.0 (9.0) with 2 cases
1. browse a SMB server on an entirely different subnet than where I am
2. browse a restricted share, which is visible to user but simply accessible by only designated users

Corresponding smb.conf entry is as follows:
```
[tmp]
    comment = Temporary directory on server
    path = /tmp
    browseable = yes
    read only = yes
    guest ok = no
    valid users = root
```
Both ended crashing the app 

## Exception
* User Action: UI Error
* Request: Application crash
* Version: 3.5.3
* OS: Linux Android 9 - 28
* Device: FP3
* Model: FP3
* Product: FP3
<details><summary><b>Crash log </b></summary><p>


java.lang.NullPointerException: Attempt to read from field 'java.lang.Object androidx.core.util.Pair.second' on a null object reference
 at com.amaze.filemanager.ui.fragments.MainFragment.lambda$loadlist$3$MainFragment(MainFragment.java:954)
 at com.amaze.filemanager.ui.fragments.-$$Lambda$MainFragment$9qQv21X8JwPlLXi_cDXfu4p-_VI.onAsyncTaskFinished(Unknown Source:8)
 at com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.onPostExecute(LoadFilesListTask.java:270)
 at com.amaze.filemanager.asynchronous.asynctasks.LoadFilesListTask.onPostExecute(LoadFilesListTask.java:67)
 at android.os.AsyncTask.finish(AsyncTask.java:695)
 at android.os.AsyncTask.access$600(AsyncTask.java:180)
 at android.os.AsyncTask$InternalHandler.handleMessage(AsyncTask.java:712)
 at android.os.Handler.dispatchMessage(Handler.java:106)
 at android.os.Looper.loop(Looper.java:193)
 at android.app.ActivityThread.main(ActivityThread.java:6718)
 at java.lang.reflect.Method.invoke(Native Method)
 at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:491)
 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)


</details>
<hr>

The null check can prevent crash, which case 1 would give the SMB connect dialog back, and case 2 will simply not respond. Not good in UX, but at least not crash - you may give user a Toast for this.

**Review by TranceLove — CHANGES_REQUESTED** [review]

**Comment by TranceLove** [comment]
build test apk

**Review by TranceLove — APPROVED** [review]
Sorry for the late reply. Checked OK on Fairphone 3 running LineageOS 16.0 (9.0).

+1.

**Pull request merged**